hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -555,15 +555,24 @@ void setDataBlockEncoderInTest(HFileDataBlockEncoder blockEncoder) {
     }
     // initialize the thread pool for opening store files in parallel..
     ThreadPoolExecutor storeFileOpenerThreadPool =
-      this.region.getStoreFileOpenAndCloseThreadPool("StoreFileOpener-"
-        + this.region.getRegionInfo().getEncodedName() + "-" + this.getColumnFamilyName());
+      this.region.getStoreFileOpenAndCloseThreadPool();
     CompletionService<HStoreFile> completionService =
       new ExecutorCompletionService<>(storeFileOpenerThreadPool);
-
+    String executingThreadNamePrefix = "StoreFileOpener-"
+      + this.region.getRegionInfo().getEncodedName() + "-" + this.getColumnFamilyName();
+    AtomicInteger threadId = new AtomicInteger(0);
     int totalValidStoreFile = 0;
     for (StoreFileInfo storeFileInfo : files) {
       // open each store file in parallel
-      completionService.submit(() -> this.createStoreFileAndReader(storeFileInfo));
+      completionService.submit(() -> {
+        Thread t = Thread.currentThread();
+        String name = t.getName();
+        Thread.currentThread().setName(executingThreadNamePrefix + "-"
```

Comment:
This is done to dynamically change the thread name as per the Store name?

Reply:
Yes